fix: error C2338: static_assert failed: 'not enough arguments provide…

yomarhub opened the pull request:
…d to format string.'

Plugin didn't compile because there was an "%s" without an argument

yomarhub committed:
fix: error C2338: static_assert failed: 'not enough arguments provided to format string.'

yomarhub committed:
refactor: rename project source directory variable and update related references to project root directory to make it work even in plugins content folder

yomarhub committed:
Correct README after refractoring

yomarhub committed:
refactor: replace comment markers with #pragma directives for better code organization

kirby561 commented:
Hey thanks for the pull request and sorry for taking so long to look at it.

The static_assert fix and using the project root for the blueprint map look good.
I have a couple issues before merging it though:
1) I like the idea of the collapsible regions but pragma region doesn't work by default in other IDE's like vs code and I don't really like how the generated sections look now, it's harder to see them:
<img width="765" height="151" alt="image" src="https://github.com/user-attachments/assets/2a02bec0-3096-4050-956f-6ef88dd31ea1" />

I think this would be better:
<img width="693" height="207" alt="image" src="https://github.com/user-attachments/assets/05aa76b6-3e46-4903-8443-b0f44209f83f" />

What if instead we just have a configurable Prefix and Suffix for generated sections so you can put in whatever region specifier you prefer if any? Then these can just get added in when filling out the generated sections and you can use whatever you want including nothing. So the config file would look like this:
```
[/Script/UmgControllerGeneratorPlugin.CodeGeneratorConfig]
ClassSuffix="Controller"
BlueprintSourceMapDirectory=""
EnableAutoReparenting=true
GeneratedMethodsPrefix="#pragma region Generated Methods Section"
GeneratedMethodsSuffix="#pragma endregion Generated Methods Section"
GeneratedIncludesPrefix="#pragma region Generated Includes Section"
GeneratedIncludesSuffix="#pragma endregion Generated Includes Section"
GeneratedLoaderPrefix="#pragma region Generated Loader Section"
GeneratedLoaderSuffix="#pragma endregion Generated Loader Section"
GeneratedPropertiesPrefix="#pragma region Generated Properties Section"
GeneratedPropertiesSuffix="#pragma endregion Generated Properties Section"
```

That way you could wrap it with whatever region specifier you wanted or even put something else there.

2) CodeGenerator.cpp:
There's a bunch of newlines added in this file, I think it got auto-formatted with different settings, mind removing? Ex: 
`if (...) { `
-> 
```
if (...)
{
```

Anyway, thanks again for the request. If you feel like making those changes I'll merge it. Otherwise I may add that at some point and we can merge then.

yomarhub committed:
refactor: move region section definitions to editor config for better flexibility

Relocated all region marker strings (e.g., #pragma region ...) from code to UCodeGeneratorConfig settings.
Allows customization of generated section delimiters via editor config.
Improves maintainability and adaptability of code generation templates.

kirby561 commented:
Hey thanks again for the contribution! I merged the PR.

I did notice an issue where UpdateHeaderFile and UpdateCppFile don't work anymore because they aren't using the configured start/end markers and so I made another commit on top that fixed that and did a few other things:
- Added support for newlines in the configured start/end markers
- Removed the don't modify manually section since it can be configured now and is included in the default
- Made the defaults match the existing style (may as well stay backwards compatible)
- Updated the readme to show how to configure #pragmas in the config file.
